### PR TITLE
Allow to create a user for a specific backend

### PR DESCRIPTION
**Pull request by nickvergessen**
Downstream 27596

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

File: lib/private/User/Manager.php [hunk 1]
```diff
@@ -321,10 +322,27 @@ public function createUser($uid, $password) {
...
 	}
 
 	/**
+	 * @param string $uid
+	 * @param string $password
+	 * @param UserInterface $backend
+	 * @return IUser|null
+	 */
+	public function createUserFromBackend($uid, $password, UserInterface $backend) {
+		$this->emit('\OC\User', 'preCreateUser', [$uid, $password]);
```
Comment (MorrisJobke) [inline lib/private/User/Manager.php]: What about all the checks that are performed in the `createUser`? Like check for existence and check for empty password and check the username for `.` or `..`? I would like to see them here as well.

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
See inline comment

File: lib/private/User/Manager.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/User/Manager.php]: Vinc had the same problem:
> regarding validation, if we expect createUserFromBackend to always provide safe values, then it's fine to not validate I guess.
> 
> Looks like the validation is mostly coming from whatever the user is inputting in the users page. That validation should happen in the users controller rather.
> 
> I'm fine merging this then. 👍

Comment (nickvergessen) [inline lib/private/User/Manager.php]: Done anyway

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**